includes/functions.php

```diff
@@ -854,3 +854,42 @@ function get_masked_wp_version() {
 
 	return implode( '.', $version );
 }
+
+/**
+ * Retrieves the IDs of the ancestors of a comment.
+ *
+ * Adaption of `get_post_ancestors` from WordPress core.
+ *
+ * @see https://developer.wordpress.org/reference/functions/get_post_ancestors/
+ *
+ * @param int|WP_Comment $comment Comment ID or comment object.
+ *
+ * @return WP_Comment[] Array of ancestor comments or empty array if there are none.
+ */
+function get_comment_ancestors( $comment ) {
+	$comment = get_comment( $comment );
+
+	// phpcs:ignore Universal.Operators.StrictComparisons.LooseEqual
+	if ( ! $comment || empty( $comment->comment_parent ) || $comment->comment_parent == $comment->comment_ID ) {
```

Comment:
The `in_array` check here and in the loop won't work because we're appending the full comment to the `$ancestors` array, while `get_post_ancestors()` only uses the post IDs.

Reply:
should we also only use IDs to be consistent?

Reply:
Probably for the best, we can grab the full comments when needed.

Reply:
done